numpy/core/__init__.py

```diff
@@ -3,9 +3,25 @@
 from .info import __doc__
 from numpy.version import version as __version__
 
+import os
+
+# on Windows NumPy loads an important OpenBLAS-related DLL
+# and the code below aims to alleviate issues with DLL
+# path resolution portability with an absolute path DLL load
+if os.name == 'nt':
+    from ctypes import WinDLL
+    from pathlib import Path
+    # convention for storing / loading the DLL from
+    # numpy/.libs/, if present
+    libs_path = Path(Path(__file__).absolute().parents[1], '.libs')
+    if libs_path.exists():
+        for filename in libs_path.glob('*openblas*dll'):
```

Comment:
could this be problematic in case of incomplete uninstalls of older versions? we may get 2 or more openblas dlls in the same place. possibly that could be a problem today as well, but I think having 2 on the search path could work while loading 2 won't

Reply:
Maybe we could issue a warning if more that one is found? It might be useful for feedback if nothing else.

Reply:
I'm not sure--I thought you could only have one `numpy\.libs\` path probed at initialization time from i.e., a wheel installed to `site-packages`?

I guess a specific example scenario would be helpful.

If it is a real problem, can we distinguish new vs. old versions of the DLL by some criterion?

Reply:
I could certainly test for the case of an old DLL somehow placed in `site-packages\numpy\.libs\` alongside the new DLL, but that seems unusually pathological?

Reply:
> alongside the new DLL, but that seems unusually pathological?

I don't think it is - package uninstall is quite unreliable, any incomplete or missing uninstall can leave files behind. `pip` won't check, it will just overwrite whatever is already there.

Reply:
Ok, I'll try testing this

Reply:
I've added an Azure step to intentionally pollute `numpy/.libs` with an older DLL from NumPy 1.14 & everything seems fine on my fork testing so I've pushed it here now.

I've also added code to emit a warning that includes the names of all DLLs if > 1 is loaded into memory from that location. You should be able to see those warnings in the Azure output for Python 3.6 32/64-bit Windows runs.